src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_StackPanel.cs

```diff
@@ -19,8 +19,109 @@ namespace Uno.UI.RuntimeTests.Tests.Windows_UI_Xaml_Controls
 #if __MACOS__
 	[Ignore("Currently fails on macOS, part of #9282 epic")]
 #endif
-	public class Given_StackPanel
+	public partial class Given_StackPanel
 	{
+		private partial class MyStackPanel : StackPanel
+		{
+			public int MeasureCount { get; private set; }
+			public int ArrangeCount { get; private set; }
+
+			public Size LastMeasureOverrideReturn { get; private set; }
+			public Size LastArrangeOverrideReturn { get; private set; }
+
+			protected override Size MeasureOverride(Size availableSize)
+			{
+				MeasureCount++;
+				return LastMeasureOverrideReturn = base.MeasureOverride(availableSize);
+			}
+
+			protected override Size ArrangeOverride(Size arrangeSize)
+			{
+				ArrangeCount++;
+				return LastArrangeOverrideReturn = base.ArrangeOverride(arrangeSize);
+			}
+		}
+
+		[TestMethod]
+		[RunsOnUIThread]
+		public async Task When_Adding_Or_Removing_Child_Should_Re_Measure()
+		{
+			var SUT = new MyStackPanel()
+			{
+				Children =
+				{
+					new Border()
+					{
+						Width = 50,
+						Height = 50,
+					}
+				}
+			};
+
+			int expectedMeasureAndArrangeCount = 0;
+
+			TestServices.WindowHelper.WindowContent = SUT;
+			await TestServices.WindowHelper.WaitForLoaded(SUT);
+			await TestServices.WindowHelper.WaitForIdle();
+			expectedMeasureAndArrangeCount++;
+			Assert.AreEqual(expectedMeasureAndArrangeCount, SUT.MeasureCount);
+			Assert.AreEqual(expectedMeasureAndArrangeCount, SUT.ArrangeCount);
+			Assert.AreEqual(50, SUT.LastMeasureOverrideReturn.Height);
+			Assert.AreEqual(50, SUT.LastArrangeOverrideReturn.Height);
+
+			SUT.Children.Add(new Border()
+			{
+				Width = 50,
+				Height = 50,
+			});
+			await TestServices.WindowHelper.WaitForRelayouted(SUT);
+			expectedMeasureAndArrangeCount++;
+			Assert.AreEqual(expectedMeasureAndArrangeCount, SUT.MeasureCount);
+			Assert.AreEqual(expectedMeasureAndArrangeCount, SUT.ArrangeCount);
+			Assert.AreEqual(100, SUT.LastMeasureOverrideReturn.Height);
+			Assert.AreEqual(100, SUT.LastArrangeOverrideReturn.Height);
+
+			SUT.Children.RemoveAt(1);
+			await TestServices.WindowHelper.WaitForRelayouted(SUT);
+			expectedMeasureAndArrangeCount++;
+			Assert.AreEqual(expectedMeasureAndArrangeCount, SUT.MeasureCount);
+			Assert.AreEqual(expectedMeasureAndArrangeCount, SUT.ArrangeCount);
+			Assert.AreEqual(50, SUT.LastMeasureOverrideReturn.Height);
+			Assert.AreEqual(50, SUT.LastArrangeOverrideReturn.Height);
+
+			SUT.Children.Remove(SUT.Children.Single());
+#if !__CROSSRUNTIME__
+			await TestServices.WindowHelper.WaitForRelayouted(SUT); // arrange is skipped (incorrectly) when finalRect is default, which is the case after removing the last child from StackPanel.
```

Comment:
This is caused by https://github.com/unoplatform/uno/blob/e774ca82a87f8361beab37b69765469b94ce6033/src/Uno.UI/UI/Xaml/UIElement.Layout.crossruntime.cs#L307-L319

This code is wrong and should be removed. If it causes any issues for Canvas, it should be re-investigated to understand the real root cause, or alternatively, special case this code path for Canvas specifically, but that sounds like a hack

Reply:
The `finalRect == default` existed before the Canvas-related fix. The Canvas-related fix (which is the check for `ICustomClippingElement`) was done after it because `finalRect == default` isn't correct in the first place.

The finalRect == `default` check was added as a performance optimization, but I don't see much value in it. I don't we will hit it that much, so it can safely be removed. Probably in a follow-up PR.

Reply:
Definitely worth doing, can you create an issue, so it is not forgotten?

Reply:
https://github.com/unoplatform/uno/issues/13672